code/object/collideshipship.cpp

```diff
@@ -711,15 +741,26 @@ void calculate_ship_ship_collision_physics(collision_info_struct *ship_ship_hit_
 		impulse_mag = -impulse_mag;
 	}
 
+	// On multi clients should this collision occur? (limiting the frequency of applying the impulse prevents bugs)
+	bool should_collide = (!MULTIPLAYER_CLIENT 
+		|| (lighter == Player_obj && heavy->type == OBJ_SHIP && timestamp_elapsed(Ships[heavy->instance].multi_client_collision_timestamp))
+		|| (heavy == Player_obj && lighter->type == OBJ_SHIP && timestamp_elapsed(Ships[lighter->instance].multi_client_collision_timestamp)));
```

Comment:
Are these the right checks? Shouldn't the timestamp be checked/set for the player ship, rather than the things they're colliding into?

Reply:
Yes, this is the correct way to do it.  I changed the design so that if multiple ships are colliding with the player, it will track the timestamp separately. 

As a verbose example for clarity: I'm about to collide with Foo and Bar?  I'll collide with Foo once and set that timestamp. Now I will wait to collide with Foo until it's time.  While I'm waiting I collide with Bar once, set that timestamp.   Foo's timestamp pops, now I can collide with Foo again but not with Bar until Bar's pops.